pysteps/nowcasts/steps.py

```diff
@@ -33,6 +33,873 @@
     DASK_IMPORTED = False
 
 
+class StepsNowcaster:
+    def __init__(self, precip, velocity, timesteps, **kwargs):
+        # Store inputs and optional parameters
+        self.precip = precip
+        self.velocity = velocity
+        self.timesteps = timesteps
+        self.n_ens_members = kwargs.get("n_ens_members", 24)
+        self.n_cascade_levels = kwargs.get("n_cascade_levels", 6)
+        self.precip_threshold = kwargs.get("precip_thr", None)
+        self.kmperpixel = kwargs.get("kmperpixel", None)
+        self.timestep = kwargs.get("timestep", None)
+        self.extrapolation_method = kwargs.get("extrap_method", "semilagrangian")
+        self.decomposition_method = kwargs.get("decomp_method", "fft")
+        self.bandpass_filter_method = kwargs.get("bandpass_filter_method", "gaussian")
+        self.noise_method = kwargs.get("noise_method", "nonparametric")
+        self.noise_stddev_adj = kwargs.get("noise_stddev_adj", None)
+        self.ar_order = kwargs.get("ar_order", 2)
+        self.velocity_perturbation_method = kwargs.get("vel_pert_method", "bps")
+        self.conditional = kwargs.get("conditional", False)
+        self.probmatching_method = kwargs.get("probmatching_method", "cdf")
+        self.mask_method = kwargs.get("mask_method", "incremental")
+        self.seed = kwargs.get("seed", None)
+        self.num_workers = kwargs.get("num_workers", 1)
+        self.fft_method = kwargs.get("fft_method", "numpy")
+        self.domain = kwargs.get("domain", "spatial")
+        self.extrapolation_kwargs = kwargs.get("extrap_kwargs", None)
+        self.filter_kwargs = kwargs.get("filter_kwargs", None)
+        self.noise_kwargs = kwargs.get("noise_kwargs", None)
+        self.velocity_perturbation_kwargs = kwargs.get("vel_pert_kwargs", None)
```

Comment:
One small stylistic thing is that, as you did with the methods, you could also make all "private" variable names start with an underscore.

Reply:
With private, do you mean everything needed under the hood and not explicitly used as an input or output? Just a check because of the quotations around private

Reply:
What I meant is that it is possible to do something like this when using this class. If `filter_kwargs` were renamed to `_filter_kwargs`, you would indicate to the user that this is not allowed.

The reason I put private in quotations is that just putting a single underscore before a variable does not enforce this. A user can choose to ignore the underscore (but a linter like pylint would warn them that this is not allowed).

It is also possible in python to start the variable name with 2 underscores (`__filter_kwargs`). In this case the python interpreter actually (somewhat) enforces the privateness of this variable, by doing something called "name mangling". This might even be preferred to the single underscores (also for the private methods). Personally I use double underscores as much as possible.

See also: https://stackoverflow.com/questions/1301346/what-is-the-meaning-of-single-and-double-underscore-before-an-object-name

```
nowcaster = StepsNowcaster(...)
nowcaster.filter_kwargs = new_filter_kwargs
```

Reply:
I want to pick this up if all the other changes you suggested are implemented. We should also check this with @dnerini to see if this does not break the PySteps rules. It is in the official documentation, but I rarely see it in the actual code.